bokeh/io.py

```diff
@@ -57,10 +57,29 @@
 #-----------------------------------------------------------------------------
 
 class _CommsHandle(object):
+
+    _json = {}
+
     def __init__(self, comms, doc, json):
+        self._cellno = None
+        try:
+            from IPython import get_ipython
+            ip = get_ipython()
+            hm = ip.history_manager
+            p_prompt = list(hm.get_tail(1, include_latest=True))[0][1]
+            self._cellno = p_prompt
+        except:
+            pass
+
         self._comms = comms
         self._doc = doc
-        self._json = json
+        self._json[id(doc)] = json
```

Comment:
http://stackoverflow.com/a/11324771/277156 says using id() may break but the default hash should work (`_json[doc] = json`)